homeassistant/components/starlink/coordinator.py

```diff
@@ -61,7 +64,8 @@ def _get_starlink_data(self) -> StarlinkData:
         status = status_data(channel_context)
         location = location_data(channel_context)
         sleep = get_sleep_config(channel_context)
-        return StarlinkData(location, sleep, *status)
+        history = history_stats(parse_samples=-1, context=channel_context)
```

Comment:
Do we really need all samples? Does `parse_samples=1` work?

Reply:
We do need all for the energy sensor. Energy value would not increase with `parse_samples=1`.

Reply:
I see, I'm not sure what the underlying calls look like but it might be worth asking in the core repo if we can get a function to give us just the data we need, for the sake of avoiding redundant data.
What do you think?

Reply:
I think it's done this way so you iterate over all the data just once and get every possible metric. But separate method just for consumption (so there is no need for that `[-1]`) would "feel" nicer, that's true!

Reply:
This method has the same structure as `status_data`. Few standalone dictionaries in a list where the last item is `PowerDict` (other "blocks" contains e.g. mean ping, obstruction times, etc.)

Reply:
I looked at the impact of adding the `history_stats` call and saw no measurable difference.